### DISCUSSION TITLE
Option Groups Handling when Multiple is disabled

### DISCUSSION THREAD

**Discussion by ctalkington-brado** [sa-si-dev/virtual-select#104] [Ideas]
As of 1.0.25, if you have a virtual-select that has groupings but does not have multiple selection enabled. when you click the main group  title, it puts the label of all the sub-options as the displayed value and sets the underlying value to the first sub-option from that group from what I can tell.

I would propose that in such a case it should use the top-level assigned value from the options array. right now i have to use the `disableOptionGroupCheckbox` to prevent the virtual select from being filled with labels of all sub-options and i have to maintain an additional item if i want to allow selecting the whole group value (which is a parent id that is sent to server to query everything in such a hierarchy)

```js
options: [
    { 
      label: 'Option Group 1',
      options: [
        { label: 'Option 1-1', value: '1' },
        ...
      ],
      value: 'use this when in single selection mode if exists',
    },
    ...
  ]
```

**Comment by sa-si-dev** [comment]
I will fix the issue, and implement your suggestion in the future release.

**Comment by sa-si-dev** [comment]
@ctalkington-brado I have fixed the above issue. And have added a new feature `useGroupValue`

It would return the group's value if all of its child options are selected.

This feature is available in the [release v1.0.26](https://github.com/sa-si-dev/virtual-select/releases/tag/v1.0.26)

**Reply by ctalkington-brado** [reply]
thanks for that. any thoughts on how groups could work in a single selection mode? seems the multiple check disables it after your last change which prevents the issues with displayed labels but ideally id love to not have to duplicate the top-level group in my options like attached.

![firefox_2022-02-22_0927_Adr1sm](https://user-images.githubusercontent.com/65624018/155164473-e30d5bc3-875b-4b20-a172-7da3e71cc0f6.png)
s

**Reply by sa-si-dev** [reply]
> thanks for that. any thoughts on how groups could work in a single selection mode? seems the multiple check disables it after your last change which prevents the issues with displayed labels but ideally id love to not have to duplicate the top-level group in my options like attached.
> 
> ![firefox_2022-02-22_0927_Adr1sm](https://user-images.githubusercontent.com/65624018/155164473-e30d5bc3-875b-4b20-a172-7da3e71cc0f6.png)
> s

I don't think this would be a general behaviour. If it is a single select, we should  allow the users to choose a single option. Not all options under a group.

**Comment by thekingofspain** [comment]
I am pretty confused on the single group options. I have not been able to get a group option to select in a single select mode.  Parameters:

```
multiple = false,
searchGroup= true or false,
disableOptionGroupCheckbox = false,
useGroupValue = true
```
DOM virtualselect.options:
```
[_NonSelectable OptionGroup_ ] {
  isDisabled: false
  isGroupTitle: true
}

[_Selectable Option_ ] {
  groupIndex: n
  isDisabled: false
  isGroupOption: true
  isGroupTitle: false
}
```

From the DOM view, I would like to have the above NonSelectable OptionGroup selectable.  My input options are only 2 layers deeps, my understanding from the comment above is that I use 3 layers (duplicating the layer 1 & 2 with basically the same option) then...

- Layer 1 is not selectable
- Layer 2 is selectable
- Layer 3 is selectable.

I think the confusion is there is no documentation how an pption that contains other options translates into _isGroupOption_ and _isGroupTitle_ values, what are the valid combinations of these values, and lastly how to set them via the given properties and methods.

There is value is having both selectable and non selectable group options (and methods to turn them on and off selectively).  As for functionality, its just classification: 

-  The higher the option level, the more general the selection is. 
-  The lower the option the the more specific the answer is

Example would be NAICS codes, where one can select a code from layer 1 to layer 6.

![image](https://github.com/sa-si-dev/virtual-select/assets/2846624/975fc7d7-ed02-4710-8c6e-ee50d7547112)
